src/main/java/io/boomerang/eventing/nats/jetstream/PubTransmitter.java

```diff
@@ -80,32 +87,114 @@ public void publish(String subject, String message)
 
     // Get NATS connection
     Connection connection = connectionPrimer.getActiveConnection();
-
     if (connection == null) {
+      storeMessage(subject, message);
       throw new NoNatsConnectionException("No connection to the NATS server!");
     }
 
     // Get Jetstream stream from the NATS server (this will also automatically create the stream if
     // not present on the server)
+    try {
+      getJetStream(connection);
+    } catch (NullPointerException npe) {
+      storeMessage(subject, message);
+    }
+
+    // Create the NATS message
+    Message natsMessage = createNATSMessage(subject, message);
+
+    // Publish the message
+    PublishAck publishAck = connection.jetStream().publish(natsMessage);
```

Comment:
This can also fail due to no connection to the server. Think that the connection can go down between the check at line 90 until this line.